unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/net/DatagramSocketTest.scala

```diff
@@ -0,0 +1,308 @@
+package org.scalanative.testsuite.javalib.net
+
+import java.net._
+
+import org.junit.Test
+import org.junit.Assert._
+import org.junit.Assume._
+
+import java.io.IOException
+
+import org.scalanative.testsuite.utils.AssertThrows.assertThrows
+
+class DatagramSocketTest {
+  @Test def constructor(): Unit = {
+    val ds = new DatagramSocket(0, InetAddress.getLocalHost())
```

Comment:
I believe that JUnit (both 4 & 5) allow a class level variable
to be declared and used.   I think it will get initialized
at class instantiation (to whatever its code says). If one
wants to be belts-and-suspenders sure, one can use a 
`@BeforeClass` annotation. There are examples in `unit-tests`.

Tests are often written in a less-than-elegant or less-than-compact style so that they are self-contained and establish and de-establish their world.  In other cases, some set up
of variables can be done either before the test or when the class is instantiated, especially if they are fetched once and then never changed.  This ensures that each test uses the same value.

In this file, it looks like `InetAddress.getLocalHose()` is called a number of times from a number of places.  If the underlying value is changing during the course of the test, something is way wrong (now, I am assuming that the 
system in use is not multi-homed (more than one interface).
If it is, we have other problems).

The "define in many places" is not _wrong_ here. Especially since this is a "first write" and, I suspect, based on other code in unit-tests. 

Author's choice, I believe.